File: upgrade.pl
use strict;
use warnings;
use File::Copy;

sub error {
	my ($message) = @_;
	print "Error: " . $message . "\n";
}

print "Old shell's directory: ";
my $old = <STDIN>;
chop $old;

my $old_path_setting = "$old/data/PATH";
if(-e $old_path_setting) {
	copy($old_path_setting, "./data/PATH") or &error("failed copy path file");
}

print "Executable Extensions:\n";
my @executable_extensions = <STDIN>;
unless(@executable_extensions == 0) {
	open(EXECUTABLE_EXTENSIONS_FILE, ">", "./data/EXECUTABLE_EXTENSIONS") or &error("cannot open executable-extensions file");
	foreach (@executable_extensions) {
		print EXECUTABLE_EXTENSIONS_FILE $_;
	}
}

File: upgrade.py
import os.path
import shutil
import sys

try:
	old = input("Old shell's directory: ")
	old_path_file = old + '/data/PATH'
	if os.path.exists(old_path_file):
		shutil.copyfile(old_path_file, './data/PATH')
except SameFileError:
	pass
except OSError:
	error('failed copy path file')

try:
	print()

	print('Executable Extensions:')
	executable_extensions = sys.stdin.readlines()
	if len(executable_extensions) != 0:
		with open('./data/EXECUTABLE_EXTENSIONS', 'w') as executable_extensions_file:
			executable_extensions_file.writelines(executable_extensions)
except IOError:
	error('cannot open executable-extensions file')
